src/scriptProfiler.cpp

```diff
@@ -1215,8 +1245,78 @@ class GameInstructionNewExpression : public game_instruction {
 
 #pragma endregion Instructions
 #endif
+
+std::filesystem::path findFilePath() {
+#if __linux__
+    char_t buffer[PATH_MAX] = { 0 };
+    ssize_t len = readlink("/proc/self/exe", buffer, sizeof(buffer)-1);
+    if (len != -1) {
+        buffer[len] = '\0';
+    }
+#else
+    char_t buffer[MAX_PATH] = { 0 };
+    DWORD len = GetModuleFileNameW(nullptr, buffer, MAX_PATH);

```

Comment:
Get the filepath of the dll instead.
Then its a very simple

`dllFilePath / "../config/parameters.json"` which will save alot of code here.
Can get rid of all the custom path string operations below, and the possiblePaths stuff (because we only have one path)

But then you'll need special code for linux to find the library path. Unless there is something in /proc that stores loaded libraries?

Reply:
`/proc/self/maps` might have it, but parsing that will be cumbersome.
But could always make that config file be windows only.

Reply:
dl_iterate_phdr on linux will walk through the loaded sharedobjects. https://linux.die.net/man/3/dl_iterate_phdr